src/include/cell_inline.h

```diff
@@ -1272,12 +1272,21 @@ __wt_dsk_cell_data_ref(WT_SESSION_IMPL *session, int page_type, void *unpack_arg
 
 /*
  * __wt_page_cell_data_ref --
- *     Set a buffer to reference the data from an unpacked cell.
+ *     Set a buffer to reference the data from an unpacked cell. If we see an overflow removed cell,
+ *     we have raced with checkpoint freeing the overflow cell. Return restart for the caller to
+ *     retry the read.
  */
 static inline int
 __wt_page_cell_data_ref(WT_SESSION_IMPL *session, WT_PAGE *page, void *unpack_arg, WT_ITEM *store)
 {
-    return (__cell_data_ref(session, page, page->type, (WT_CELL_UNPACK_COMMON *)unpack_arg, store));
+    WT_CELL_UNPACK_COMMON *unpack;
+
+    unpack = (WT_CELL_UNPACK_COMMON *)unpack_arg;
```

Comment:
I don't like the typeless parameter here, but I suspect it's justified due to the performance path. This cast will probably be optimized away by the compiler (I know it's not new).

I'm tempted to ask for this to be expanded into three different functions, this one that takes a WT_CELL_UNPACK_COMMON, one that accepts a WT_CELL_UNPACK_KV and another that accepts a WT_CELL_UNPACK_ADDR. The two latter functions (would be inline) would just call this one with a cast to the WT_CELL_UNPACK_COMMON version.

I'm guessing the answer to my prior question about all callers handling the new WT_RESTART return would influence whether to pursue this.

Reply:
I don't see anywhere this function is called with WT_CELL_UNPACK_ADDR